src/plugins/data_views/common/data_views/data_view_lazy.ts

```diff
@@ -464,19 +471,20 @@ export class DataViewLazy extends AbstractDataView {
     );
   }
 
-  async getComputedFields({ fieldName = ['*'] }: { fieldName: string[] }) {
+  getComputedFields({ fieldName = ['*'] }: { fieldName: string[] }) {
     const scriptFields: Record<string, estypes.ScriptField> = {};
+    if (!this.fields) {
+      return {
+        scriptFields,
+        docvalueFields: [] as Array<{ field: string; format: string }>,
+        runtimeFields: {},
+      };
+    }
 
-    const fieldMap = (
-      await this.getFields({
-        fieldName,
-        fieldTypes: ['date', 'date_nanos'],
-        scripted: false,
-        runtime: false,
-      })
-    ).getFieldMap();
-
-    const docvalueFields = Object.values(fieldMap).map((dateField) => {
+    // Date value returned in "_source" could be in a number of formats
+    // Use a docvalue for each date field to ensure standardized formats when working with date fields
+    // dataView.flattenHit will override "_source" values when the same field is also defined in "fields"
```

Comment:
I just copied code from DataView to here, to make this functional. which it does. alternatively we could also turn a DataViewLazy into a regular DataView without more fields being loaded. would also work.